modules/swffg-main.js

```diff
@@ -600,21 +609,83 @@ Hooks.once("ready", async () => {
         CONFIG.logger.error(err);
       }
     }
+    // migrate embedded items
+    if (isAlpha || currentVersion === "null" || currentVersion === '') {
+      ui.notifications.info(`Migrating Star Wars FFG System Deep Embedded Items`)
+      CONFIG.logger.debug('Migrating Star Wars FFG System Deep Embedded Items')
+
+      // start with items on actors
+      game.actors.forEach((actor) => {
+        actor.items.forEach((item) => {
+          if (["weapon", "armour"].includes(item.type)) {
+            // iterate over attachments and modifiers on the item
+            item.system.itemmodifier.forEach((modifier) => {
+              if (modifier.hasOwnProperty('data')) {
+                modifier.system = modifier.data;
+                delete modifier.data;
+              }
+            });
+
+            item.system.itemattachment.forEach((attachment) => {
+              if (attachment.hasOwnProperty('data')) {
+                attachment.system = attachment.data;
+                delete attachment.data;
+              }
+            });
+            // copy the item, so we can delete the ID field (we can't update if we include an ID)
+            let item_migrated = structuredClone(item);
+            let item_id = item._id;
+            delete item_migrated._id;
+            // persist the changes to the DB
+            actor.items.filter(i => i._id === item_id)[0].update(item_migrated);
```

Comment:
Can we not just call `item.update(item_migrated)` here?

Reply:
nope, it complains that you're trying to modify the ID (even though they match), and the ID is a read-only field, so you can't even delete it

Reply:
here's the error when it's the way you asked about
![image](https://user-images.githubusercontent.com/4709746/192058138-f2665251-5f44-43dc-86ff-7783850f9c25.png)

Reply:
Is that using the item_migrated object as the param though, after you delete _id from it?

Reply:
Oh, no, it's simply `item.update(item)`. I'll try your suggestion - misunderstood the first... two times :p